### PR TITLE
Avoid multiple time-series by filtering the carousel options

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: #12395 

The widget type options in the carousel are filtered to avoid creating multiple time-series widgets:

filter-timeseries-options

Bonus:
- Add a widget with the `Enter` key
- Fix reloading widgets view with `Enter` if not focused

For acceptance:
1. Add two non time-series widgets
2. Convert a current widget to time-series
3. Check that the other widget can not be changed to time-series

### CODE REVIEW

File: lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js [hunk 1]
```diff
@@ -65,6 +70,26 @@ module.exports = CoreView.extend({
     return this;
   },
 
+  clean: function () {
```
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js]: Let's try to add clean functions at the end of the classes please 🙏

File: lib/assets/core/javascripts/cartodb3/editor/widgets/widgets-form/widgets-form-view.js [hunk 2]
```diff
@@ -47,8 +48,21 @@ module.exports = CoreView.extend({
   },
 
   _renderCarousel: function () {
+    var containsTimeSeries = false;
```
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/editor/widgets/widgets-form/widgets-form-view.js]: I miss the tests validating this @Jesus89  💃 .

File: lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js (hunk 1 above)
Comment (Jesus89) [reply lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js]: Ok!

File: lib/assets/core/test/spec/cartodb3/editor/widgets/widgets-form/widgets-form-view.spec.js [hunk 3]
```diff
@@ -126,6 +127,69 @@ describe('editor/widgets/widgets-form/widgets-form-view', function () {
...
+        userActions: this.userActions,
+        querySchemaModel: this.querySchemaModel,
+        configModel: {},
+        widgetDefinitionModel: widgetDefinitionModel2,
+        userModel: this.userModel,
+        modals: this.modals
+      });
+    });
+
+    it('should not include time-series in carousel options if there is a time-series widget', function () {
```
Comment (xavijam) [inline lib/assets/core/test/spec/cartodb3/editor/widgets/widgets-form/widgets-form-view.spec.js]: Why not adding both tests under a proper describe?

Comment (xavijam) [reply lib/assets/core/test/spec/cartodb3/editor/widgets/widgets-form/widgets-form-view.spec.js]: With that, you could create another view in the beforeEach of that describe and not adding it in the "main" beforeEach. Doing that, we could remove an unnecessary view for the rest of the tests.
Also, it is great to clean the view at the end of the tests. 

```js
afterEach(function () {
  view.clean();
});
```

Comment (Jesus89) [reply lib/assets/core/test/spec/cartodb3/editor/widgets/widgets-form/widgets-form-view.spec.js]: There is already a describe(.'_getFilteredDataTypes') with its "local" beforeEach, so I am not using the "main" beforeEach. I will add the afterEach method for cleaning.

Comment (Jesus89) [reply lib/assets/core/test/spec/cartodb3/editor/widgets/widgets-form/widgets-form-view.spec.js]: Btw, I will add one more test.

File: lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js [hunk 4]
```diff
@@ -65,6 +70,21 @@ module.exports = CoreView.extend({
...
 
+  _initBinds: function () {
+    $(document).bind('keydown', this._onKeyDown);
+  },
+
+  _disableBinds: function () {
+    $(document).unbind('keydown', this._onKeyDown);
+  },
+
+  _onKeyDown: function (e) {
```
Comment (rubenmoya) [inline lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js]: I'd change this to `event` since it's more explicit

Comment (Jesus89) [reply lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js]: I have used the same style seen in the code (about ~150 "function (e)" vs ~30 "function (event)")

Comment (rubenmoya) [reply lib/assets/core/javascripts/cartodb3/components/modals/add-widgets/add-widgets-view.js]: I know, I've been _boyscouting_ it slowly, we should stop using things like `e`, `m`, `mdl`. `e` could be anything, could be `event`, could be `error`...

**Comment by nobuti** [comment]
Works like a charm ✨

**Comment by nobuti** [comment]
🚀

**Pull request merged**